content/gotraining-2019-05-melbourne.slide

```diff
@@ -645,6 +645,28 @@ If `master` on upstream has advanced in the meantime:
 	}
 
 
+* Embedding - Structs
+
+"When we embed a type, the methods of that type become methods of the outer type, but when they are invoked the receiver of the method is the inner type, not the outer one." - [[https://golang.org/doc/effective_go.html#embedding][Effective Go]]
+
+- The same applies for all fields within an embedded type
+
+
+* Precedence when Embedding
+
+Embedding types introduces the possiblity for naming conflicts. Go follows some simple rules to help resolve this:
+
+- A field or method "X" will hide any other field or method "X" found in a more deeply nested part of the type
+- If a field or method appears at the same nesting level, it will return an error
+
+
+* Examples
+
+- [[https://goplay.space/#t4oJw-_2mph][Embedded Structs]]
```

Comment:
I'd use a sample with a type hierarchy, rather than types "on the same level" - e.g. Pet and Cat: https://goplay.space/#Uggvx8Cb5-p